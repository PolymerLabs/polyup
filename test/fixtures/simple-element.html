<link rel='import' href='../polymer/polymer.html'>
<link rel='import' href='../paper-spinner/paper-spinner.html'>
<link rel='import' href='../core-ajax/core-ajax.html'>

<polymer-element name='simple-element'
                 attributes='intAttr boolAttr objAttr arrayAttr' tabindex='0'>
  <template>
    <style>
      :host {
        display: block;
        margin: 20px;
      }
    </style>
    <div for="{{strAttr}}" style="background-color: {{strAttr}};">
      <a href='http://example.com/{{strAttr}}'>Hello there</a>
    </div>
    <input>
    <label on-click='{{labelClicked}}' for='{{strAttr}}'>Select one:</label>
    <select id='{{strAttr}}' value='{{strAttr}}' hidden?='{{!boolAttr}}'>
      <option value='1'>One</option>
      <option value='2' data-secret='{{intAttr}}'>Two</option>
    </select>
    <core-ajax id='ajax'></core-ajax>
    <template if='{{boolAttr}}'>
      <div>{{intAttr}} times I tell you {{strAttr}}.</div>
      <div>{{intAttr}} times I tell you {{strAttr}}</div>
      <div>Yes {{intAttr}} times I tell you {{strAttr}}</div>
      <div>Yes {{intAttr}} times I tell you {{strAttr}}.</div>
    </template>
  </template>
  <script>
    // this is a comment
    Polymer('simple-element', {
      publish: {
        /**
         * The most important attribute in the world.
         */
        intAttr: {
          value: 0,
          reflect: true
        }
      },
      computed: {
        'url': '"/api/v" + intVal + "/get"',
        auto: 'shouldAuto(intVal, boolVal)',
        'params': '{ok: boolVal, good: isGood(intVal)}'
      },
      observe: {
        'intAttr x z.v.k': 'somethingsUp',
      },
      objAttr: {},
      arrayAttr: [],
      boolAttr: false,
      ready: function() {
        console.log('ready!')
      },
      domReady: function() {
        console.log('dom is ready!');
      },
      boolAttrChanged: function(oldVal, newVal) {
        console.log('changed from ', oldVal, ' to ', newVal);
      },
      somethingsUp: function(oldVal) {
        console.log('something changed from', oldVal)
      },
      shouldAuto: function(intVal, boolVal) {
        return intVal + boolVal;
      },
      isGood: function(intVal) {
        return intVal > 1;
      }
    });
  </script>
</polymer-element>
