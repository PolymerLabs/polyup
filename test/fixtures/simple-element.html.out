<link rel="import" href="../polymer/polymer.html">
<link rel="import" href="../paper-spinner/paper-spinner.html">
<link rel="import" href="../core-ajax/core-ajax.html">

<!--
    This test is still incomplete, we still need to fixup the core-ajax import.
 -->
<dom-module id="simple-element">
  <style>
      :host {
        display: block;
        margin: 20px;
      }
    </style>
  <template>
    <div>Hello there</div>
    <iron-ajax id="ajax"></iron-ajax>
    <template is="dom-if" if="{{boolAttr}}">
      <div>{{intAttr}} times I tell you hello.</div>
    </template>
  </template>
</dom-module>
<script>
  // this is a comment
  Polymer({
    is: 'simple-element',
    properties: {
      arrayAttr: {
        type: Array,
        value: function () {
          return [];
        },
        notify: true
      },
      auto: { computed: 'shouldAuto(int_val, bool_val)' },
      boolAttr: {
        type: Boolean,
        value: false,
        notify: true,
        observer: 'boolAttrChanged'
      },
      /**
       * The most important attribute in the world.
       */
      intAttr: {
        type: Number,
        value: 0,
        notify: true,
        observer: 'somethingsUp',
        reflectToAttribute: true
      },
      objAttr: {
        type: Object,
        value: function () {
          return {};
        },
        notify: true
      },
      params: { computed: 'computeParams(bool_val, int_val)' },
      url: { computed: 'computeUrl(int_val)' },
      x: { observer: 'somethingsUp' }
    },
    ready: function () {
      console.log('ready!');
      console.log('dom is ready!');
    },
    boolAttrChanged: function (newVal, oldVal) {
      console.log('changed from ', oldVal, ' to ', newVal);
    },
    somethingsUp: function (_, oldVal) {
      console.log('something changed from', oldVal);
    },
    shouldAuto: function (int_val, bool_val) {
      return int_val + bool_val;
    },
    isGood: function (int_val) {
      return int_val > 1;
    },
    hostAttributes: { tabindex: '0' },
    computeParams: function (bool_val, int_val) {
      return {
        ok: bool_val,
        good: this.isGood(int_val)
      };
    },
    computeUrl: function (int_val) {
      return '/api/v' + int_val + '/get';
    }
  });
</script>
